checks_windows.go

```diff
@@ -0,0 +1,410 @@
+
+func (c cond) SecurityPolicy() (bool, error) {
+	var desiredString string
+
+	// If the passed key is one we know is in the registry, just wrap
+	// RegistryKey.
+	if regKey, ok := secpolToKey[c.Key]; ok {
+		return cond{
+			Key:   regKey,
+			Value: c.Value,
+		}.RegistryKey()
+	}
+
+	// Otherwise, we're going to grab and parse secedit output :/
+	seceditOutput, err := getSecedit()
+	if err != nil {
+		return false, err
+	}
+
+	re := regexp.MustCompile("(?m)[\r\n]+^.*" + c.Key + ".*$")
+	output := strings.TrimSpace(string(re.Find([]byte(seceditOutput))))
+	if output == "" {
+		return false, errors.New("SecurityPolicy item not found")
+	}
+
+	if c.Key == "NewAdministratorName" || c.Key == "NewGuestName" {
+		// These two are strings, not numbers, so they have ""
+		desiredString = c.Key + " = " + c.Value
+	} else if c.Key == "MinimumPasswordAge" ||
+		c.Key == "MinimumPasswordLength" ||
+		c.Key == "LockoutDuration" ||
+		c.Key == "ResetLockoutCount" ||
+		c.Key == "MaximumPasswordAge" ||
+		c.Key == "LockoutBadCount" {
+
+		// These keys are integers, and support ranges.
+		var outputResult, err = strconv.Atoi(strings.Split(output, " = ")[1])
+		if err != nil {
+			return false, err
+		}
+
+		if strings.Contains(c.Value, "-") {
+			splitVal := strings.Split(c.Value, "-")
+			if len(splitVal) != 2 {
+				fail("Malformed range value:", c.Value)
+				return false, errors.New("Invalid c.Value range")
+			}
+			intLow, err := strconv.Atoi(splitVal[0])
+			if err != nil {
+				fail(splitVal[0] + " is not a valid integer for SecurityPolicy check")
+				return false, err
+			}
+			intHigh, err := strconv.Atoi(splitVal[1])
+			if err != nil {
+				fail(splitVal[1] + " is not a valid integer for SecurityPolicy check")
+				return false, err
+			}
+			if intLow <= outputResult && outputResult <= intHigh {
+				return true, nil
+			}
+		} else {
+			desiredValue, err := strconv.Atoi(c.Value)
+			if err != nil {
+				fail(c.Value + " is not a valid integer for SecurityPolicy check")
+				return false, errors.New("Invalid c.Value")
+			}
+			if outputResult == desiredValue {
+				return true, nil
+			}
+		}
+	} else {
+		desiredString = c.Key + " = " + c.Value
+	}
+
+	return output == desiredString, nil
+}
+
+func (c cond) ServiceStartup() (bool, error) {
+	var startupNumber string
+	switch c.Value = strings.ToLower(c.Value); c.Value {
+	case "automatic":
+		startupNumber = "2"
+	case "manual":
+		startupNumber = "3"
+	case "disabled":
+		startupNumber = "4"
+	default:
+		fail("Unknown startup type '"+c.Value+"' for", c.Name)
+		return false, errors.New("Unknown status type found for " + c.Name)
+	}
+	serviceKey := `HKLM\SYSTEM\CurrentControlSet\Services\` + c.Name + `\Start`
+	return cond{
+		Key:   serviceKey,
+		Value: startupNumber,
+	}.RegistryKey()
+}
+
+func (c cond) ServiceUp() (bool, error) {
+	serviceStatus, err := getLocalServiceStatus(c.Name)
+	return serviceStatus.IsRunning, err
+}
+
+func (c cond) ShareExists() (bool, error) {
+	return cond{
```

Comment:
Should this also be a new issue to convert it away from CommandOutput ?

Reply:
For sure, we ideally want to migrate all of these checks away from using Command*. A new issue/PR would be great.